code/modules/clothing/head/helmet.dm

```diff
@@ -17,6 +17,56 @@
 	hitsound = list('sound/items/misc/balloon_small-hit.ogg')
 	flashbang_protection = TRUE
 
+	var/obj/item/holochip/holochip
+
+/obj/item/clothing/head/helmet/Destroy()
+	QDEL_NULL(holochip)
+	return ..()
+
+/obj/item/clothing/head/helmet/equipped(mob/user, slot)
+	if(holochip && slot == SLOT_HEAD)
+		if(user.hud_used) //NPCs don't need a map
+			user.hud_used.init_screen(/atom/movable/screen/holomap)
+		holochip.add_action(user)
+		holochip.update_freq(holochip.frequency)
+	..()
+
+/obj/item/clothing/head/helmet/dropped(mob/user)
+	if(holochip)
+		holochip.remove_action(user)
+		holochip.deactivate_holomap()
+		if(SSholomaps.holochips[num2text(holochip.frequency)] && (holochip in SSholomaps.holochips[num2text(holochip.frequency)]))
+			SSholomaps.holochips[num2text(holochip.frequency)] -= holochip
```

Comment:
`92 Runtime in helmet.dm:38 : type mismatch: null -= Holomap chip (/obj/item/holochip/ert)`